### DISCUSSION TITLE
Serialize JSON to custom buffer

### DISCUSSION THREAD

**Discussion by TheStormN** [nlohmann/json#2711] [Q&A]
Hello,

I'm trying to find a way to `dump()` nlohmann::json object to custom buffer. Currently the only way is to first dump to `std::string` and then copy that string to target buffer. I would like to avoid that copy if possible. Is there a way to do that?

**Comment by zethon** [comment]
bump

**Comment by falbrechtskirchinger** [comment]
No. While the library does have (some of) the necessary abstractions, they are not (currently) part of the public API.
